pylint/message/message_id_store.py

```diff
@@ -101,18 +102,24 @@ def _raise_duplicate_msgid(symbol: str, msgid: str, other_msgid: str) -> NoRetur
         )
         raise InvalidMessageError(error_message)
 
+    @functools.lru_cache()
     def get_active_msgids(self, msgid_or_symbol: str) -> List[str]:
-        """Return msgids but the input can be a symbol."""
-        # Only msgid can have a digit as second letter
-        is_msgid: bool = msgid_or_symbol[1:].isdigit()
-        msgid = None
-        if is_msgid:
+        """Return msgids but the input can be a symbol.
+
+        The cache has no limit as its size will likely stay minimal. For each message we store
+        about 1000 characters, so even if we would have 1000 messages the cache would only
+        take up ~= 1 Mb.
+        """
+        msgid: Optional[str]
+        symbol: Optional[str]
+        if msgid_or_symbol[1:].isdigit():
+            # Only msgid can have a digit as second letter
             msgid = msgid_or_symbol.upper()
```

Comment:
```suggestion
            msgid: Optional[str] = msgid_or_symbol.upper()
```

Don't know why I didn't think of this sooner, but this should fix it (or this change but then on L118. We don't need an extra line, we just need to add typing to the call that casts either to `str` instead of `Optional[str]`.

Reply:
Well I did that at first but then `` self.__msgid_to_symbol.get(msgid)`` expects a string and not a null-able string.

Reply:
Why did we remove the `msgid = None` line?
```python
    def get_active_msgids(self, msgid_or_symbol: str) -> List[str]:
        """Return msgids but the input can be a symbol."""
        # Only msgid can have a digit as second letter
        msgid = None
        if msgid_or_symbol[1:].isdigit():
            msgid = msgid_or_symbol.upper()
            symbol = self.__msgid_to_symbol.get(msgid)
        else:
            msgid = self.__symbol_to_msgid.get(msgid_or_symbol)
            symbol = msgid_or_symbol
        if msgid is None or symbol is None or not msgid or not symbol:
            error_msg = f"No such message id or symbol '{msgid_or_symbol}'."
            raise UnknownMessageError(error_msg)
        return self.__old_names.get(msgid, [msgid])
```

Gives not errors for me with `mypy --strict`. `mypy` infers that `msgid` is `Optional[str]` I think.

Reply:
I suppose the type hint is only for type checker when the assignment is really executed at run time and impact performance.

Reply:
That's true!